packages/vite/src/node/build.ts

```diff
@@ -706,7 +695,7 @@ async function buildEnvironment(
 
       const format = output.format || 'es'
       const jsExt =
-        environment.config.consumer === 'server' || libOptions
+        (ssr && !isSsrTargetWebworkerEnvironment) || libOptions
```

Comment:
`ssr` is `environment.config.consumer === 'server'`.
https://github.com/vitejs/vite/blob/ae6895869157e48b32088f0a1f85d2fddb2d713f/packages/vite/src/node/build.ts#L549